src/shared/bus-util.c

```diff
@@ -1295,6 +1295,13 @@ int bus_connect_transport(BusTransport transport, const char *host, bool user, s
         assert_return((transport == BUS_TRANSPORT_LOCAL) == !host, -EINVAL);
         assert_return(transport == BUS_TRANSPORT_LOCAL || !user, -EOPNOTSUPP);
 
+        if (!user && sd_booted() <= 0) {
```

Comment:
Even if `host` is set, this check could be true. But I think it's not necessary to have systemd running locally for remote transport.

Reply:
Yeah, so I've asked in telegram few days ago. @poettering saids
> It requires systemd to be installed since it needs that stdio-to-dbus bridge to work

Reply:
`systemd-stdio-bridge` is executed on the remote side, not local. So I don't think there's any requirement locally. Remotely, systemd needs to be installed, so that `systemd-stdio-bridge` is present, and dbus needs to be running, but we cannot check this locally.

Reply:
It seems that the necessary condition is `systemd` is running on the _remote_ host. Not local.

Reply:
Ah, we posted almost the same time. Thank you for the clarification. I've confirmed that.